src/omlt/neuralnet/layers/partition_based.py

```diff
@@ -106,7 +114,10 @@ def output_node_block(b, *output_index):
         expr += bias
 
         lb, ub = compute_bounds_on_expr(expr)
-        assert lb is not None and ub is not None
+        if lb is None:
+            raise ValueError("Expression is unbounded below.")
+        if ub is None:
+            raise ValueError("Expression is unbounded above.")
```

Comment:
What expression? More context should be provided to users.

Reply:
I think the auxiliary variables introduced in partition formulation will always be bounded, and wonder if we really need these error messages.